# Select xRX200 and xRX300/330 firmware if CONFIG_XRX200_PHY_FW (built-in gphy firmware selection module) is enabled. Macro CONFIG_XRX200_PHY_FW must be disabled if gphy fw is loaded via userspace bootup scripts.
# Add an option to indicate if IMAGE_EXTRA_CHECKS are enabled or not in gphy_firmware image. This is used by switch_api code. This will be enabled via userspace macro: CONFIG_KERNEL_LANTIQ_IMAGE_EXTRA_CHECKS

diff --git a/arch/mips/lantiq/Kconfig b/arch/mips/lantiq/Kconfig
--- a/arch/mips/lantiq/Kconfig
+++ b/arch/mips/lantiq/Kconfig
@@ -169,4 +169,9 @@ config XRX200_PHY_FW
 	bool "XRX200 PHY firmware loader"
 	depends on SOC_XWAY
 
+config LANTIQ_IMAGE_EXTRA_CHECKS
+	bool "Extra check fields on PHY firmware header"
+	default n
+	depends on SOC_XWAY
+
 endif
diff --git a/firmware/Makefile b/firmware/Makefile
--- a/firmware/Makefile
+++ b/firmware/Makefile
@@ -134,9 +134,16 @@ fw-shipped-$(CONFIG_USB_SERIAL_KEYSPAN_P
 fw-shipped-$(CONFIG_USB_SERIAL_XIRCOM) += keyspan_pda/xircom_pgs.fw
 fw-shipped-$(CONFIG_USB_VICAM) += vicam/firmware.fw
 fw-shipped-$(CONFIG_VIDEO_CPIA2) += cpia2/stv0672_vp4.bin
+ifdef CONFIG_XRX200_PHY_FW
 fw-shipped-$(CONFIG_SOC_XWAY) += lantiq/vr9_phy11g_a1x.bin
 fw-shipped-$(CONFIG_SOC_XWAY) += lantiq/vr9_phy11g_a2x.bin
 fw-shipped-$(CONFIG_SOC_XWAY) += lantiq/vr9_phy22f_a1x.bin
+fw-shipped-$(CONFIG_SOC_XWAY) += lantiq/ltq_fw_PHY22F_IP_1v5_xRx3xx_A21_RC434.bin
+fw-shipped-$(CONFIG_SOC_XWAY) += lantiq/ltq_fw_PHY11G_IP_1v5_xRx3xx_A21_R8434.bin
+endif
+ifdef CONFIG_XRX500_PHY_FW
+fw-shipped-$(CONFIG_SOC_GRX500) += lantiq/phy11g_ip_BE.bin
+endif
 fw-shipped-$(CONFIG_YAM) += yam/1200.bin yam/9600.bin
 
 fw-shipped-all := $(fw-shipped-y) $(fw-shipped-m) $(fw-shipped-)
